### PR TITLE
Support user-provided client certs for Vault

### PR DESCRIPTION
**Pull request by atye**
# Description

- Supports user-provided certificates for connecting to Vault
- Certificates are provided in base64 format in the CR
- Storage-service deployment is installed dynamically and removed from deployment.yaml manifest
- Fixes an issue installing v1 by having a no-op for installing Authorization CRDs if v1
- Adds openshift unit tests for Ingress to help coverage

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

New unit tests covering v1 and v2 flow and passing in vault client certificates.

Manual installation of v2 with self-signed certs for Vault and user-proved certs to validate the Vault server. Manual installation of v1.

### CODE REVIEW

File: pkg/modules/authorization.go [hunk 1]
```diff
@@ -631,6 +633,543 @@ func AuthorizationServerDeployment(ctx context.Context, isDeleting bool, op util
...
+		switch component.Name {
+		case AuthProxyServerComponent:
+			image = component.StorageService
+		}
+	}
+
+	replicas := int32(1)
+	// set deployment
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
```
Comment (shaynafinocchiaro) [inline pkg/modules/authorization.go:679]: Missing TypeMeta for deployment

File: pkg/modules/authorization.go [hunk 2]
```diff
@@ -631,6 +633,543 @@ func AuthorizationServerDeployment(ctx context.Context, isDeleting bool, op util
...
+			Secret: &corev1.SecretProjection{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: "storage-service-selfsigned-tls",
+				},
+			},
+		})
+	}
+
+	// set deployment
+	deployment := appsv1.Deployment{
```
Comment (shaynafinocchiaro) [inline pkg/modules/authorization.go:890]: I know there are some differences between V1 and V2 deployments, but is there a way to reuse one deployment struct?

File: pkg/modules/authorization.go [hunk 3]
```diff
@@ -631,6 +633,543 @@ func AuthorizationServerDeployment(ctx context.Context, isDeleting bool, op util
...
+		return fmt.Errorf("converting storage-service-selfsigned issuer json to yaml: %v", err)
+	}
+
+	// create/delete issuer
+	err = applyDeleteObjects(ctx, ctrlClient, string(issuerYaml), isDeleting)
+	if err != nil {
+		return err
+	}
+
+	certificate := &certificate.Certificate{
```
Comment (shaynafinocchiaro) [inline pkg/modules/authorization.go:1121]: Can the `createSelfSignedCertificate ` function be reused here?

File: pkg/modules/authorization.go (hunk 1 above)
Comment (atye) [reply pkg/modules/authorization.go:683]: Fixed.

File: pkg/modules/authorization.go (hunk 2 above)
Comment (atye) [reply pkg/modules/authorization.go]: Refactored to use a common deployment struct that is modified by the callers of the v1 and v2 flow.

File: pkg/modules/authorization.go (hunk 3 above)
Comment (atye) [reply pkg/modules/authorization.go]: Refactored to use this func.

File: pkg/modules/authorization.go (hunk 1 above)
Comment (atye) [reply pkg/modules/authorization.go]: Manually deployed Authorization to verify change.

File: pkg/modules/authorization.go (hunk 2 above)
Comment (atye) [reply pkg/modules/authorization.go]: Manually deployed Authorization to verify change.

File: pkg/modules/authorization.go (hunk 3 above)
Comment (atye) [reply pkg/modules/authorization.go]: Manually deployed Authorization v1 and v2 to verify change.

**Review by shaynafinocchiaro — APPROVED** [review]

File: pkg/modules/testdata/cr_auth_proxy_vault_cert.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: csm-config-params
+  namespace: authorization
+data:
+  csm-config-params.yaml: |
+    CONCURRENT_POWERFLEX_REQUESTS: 10
+    LOG_LEVEL: debug
```
Comment (EvgenyUglov) [inline pkg/modules/testdata/cr_auth_proxy_vault_cert.yaml:117]: Do we need to add capacity poll interval here? Same for other yamls

Comment (atye) [reply pkg/modules/testdata/cr_auth_proxy_vault_cert.yaml:117]: It doesn't break any existing tests or tests that I added. @KerryKovacevic might add a yaml with it for her work.

**Pull request merged**